### PR TITLE
Frontend: create wallet with advanced backup options

### PR DESCRIPTION
**Pull request by thisconnect**
and added support for skip microSD card and 12 words config.

### CODE REVIEW

File: frontends/web/src/routes/device/bitbox02/bitbox02.tsx [hunk 1]
```diff
@@ -102,7 +105,8 @@ class BitBox02 extends Component<Props, State> {
         this.setState({ unlockOnly: false });
       }
       if (status === 'seeded') {
-        this.setState({ appStatus: 'createWallet' });
+        console.log('when does this happen: seeded???????', this.state.createOptions);
```
Comment (thisconnect) [inline frontends/web/src/routes/device/bitbox02/bitbox02.tsx:108]: Found it, this happens if the BBApp is restarted during setup. Currently it just starts the create-wallet process again. But I think if the user has choosen 12 words or skip sdcard this info is lost...

The user probably would only restart if there was some kind of issue, or if they just forgot (went to lunch)...

Maybe better to go back to setup choices?

cc @benma WDYT?

Comment (benma) [reply frontends/web/src/routes/device/bitbox02/bitbox02.tsx:108]: this status event is fired after the bitbox02 password was defined, app restart not necessary. so you can't really go back to the setup choices here. that would rip the user out of the regular setup flow.

Comment (thisconnect) [reply frontends/web/src/routes/device/bitbox02/bitbox02.tsx:108]: ah true 🙈

Comment (thisconnect) [reply frontends/web/src/routes/device/bitbox02/bitbox02.tsx:108]: In this case, I suggest we just remove `this.setState({ appStatus: 'create-wallet' });` here for the following reason:

appStatus is already  'create-wallet'

restarting the app after passsword was set can then go back to setup choices, which is better for the reason I mentioned before:

>  Currently it just starts the create-wallet process again. But I think if the user has choosen 12 words or skip sdcard this info is lost

Comment (thisconnect) [reply frontends/web/src/routes/device/bitbox02/bitbox02.tsx:108]: oups, I noticed my testing is a bit wrong, just reloading the frontend behaves different than restarting servewallet (obviously). Restarting frontend after password goes to set wallet again. Restarting servewallet goes to setup choices, (which is what I proposed anyway).

File: frontends/web/src/routes/device/bitbox02/setup/wallet-create.tsx [hunk 2]
```diff
@@ -43,10 +47,12 @@ export const CreateWallet = ({
   const [status, setStatus] = useState<TCreateWalletStatus>('intro');
   const [errorText, setErrorText] = useState('');
 
+  console.log('CreateWallet', backupType, backupSeedLength);
```
Comment (benma) [inline frontends/web/src/routes/device/bitbox02/setup/wallet-create.tsx:50]: remove again?

**Review by benma — APPROVED** [review]
didn't do a code review but tested all combinations:

- sdcard, 12 words
- sdcard, 24 words
- skip sdcard, 12 words
- skip sdcard, 24 words
- 12 word feature disabled
- skip sdcard feature disabled

Everything works and I really like how you solved putting it into the advanced options! slick


Here are some small things I noticed:

1. Manage Device -> Show recovery words. 

The dialog says 'You will be presented with **24** recovery words, which form a backup of your wallet. Write them down on paper.' Should be reworded to "You will be presented with your recovery words" or similar, as it can also be 12 words.

2. if a feature is disabled it says so in red text. visually it would be better to move that red text to a new line

3. the whole label "Skip microSD card ..." and "Create 12-word ..." labels cannot be selected?! everything should be selectable so it can be copy/pasted etc.

4. please update the changelog :rocket:

File: frontends/web/src/routes/device/bitbox02/setup/wallet-create.tsx (hunk 2 above)
Comment (thisconnect) [reply frontends/web/src/routes/device/bitbox02/setup/wallet-create.tsx]: removed

**Comment by thisconnect** [comment]
thanks for testing 🙏 

changed some more text that mentioned 24 words (in passphrase ino) and will update locize before merging this PR. 

> 1. Manage Device -> Show recovery words.

also addressed feedback 2. 3. and 4.

**Review by Beerosagos — APPROVED** [review]
utACK! Just a small nit in a comment below. Very nice :fire:

File: frontends/web/src/routes/device/bitbox02/setup/choose.tsx [hunk 3]
```diff
@@ -14,58 +14,208 @@
...
+
+export type TWalletCreateOptions = {
+  withMnemonic: boolean;
+  with12Words: boolean;
+};
 
 type Props = {
-  onSelectSetup: (option: 'create-wallet' | 'restore-sdcard' | 'restore-mnemonic') => void;
+  onSelectSetup: (
+    type: 'create-wallet' | 'restore-sdcard' | 'restore-mnemonic',
```
[the remark this answers is not in the document]
Comment (thisconnect) [reply frontends/web/src/routes/device/bitbox02/setup/choose.tsx]: thanks fixed

**Comment by thisconnect** [comment]
I did some more testing and couldn't reproduce the issue @Beerosagos had once.

**Pull request merged**